### ISSUE TITLE
RBXM Parsing Error: RBXM Ser/De Error: Unknown property type `32`

### ISSUE THREAD

**Issue by GlowingUmbreon** [CraftSpider/rbxm-rs#4]
It appears roblox has added a new type and now files are failing to (de)serialise.

**Comment by CraftSpider** [comment]
Thank you for the heads-up, I'll try to take a look and get a fix pushed in the next day or so.

**Comment by GlowingUmbreon** [comment]
It appears this new property is the `Font` type https://github.com/RobloxAPI/spec/blob/master/formats/rbxl.md#font

**Comment by CraftSpider** [comment]
I've added the new type as `FontFace`, I'm currently working on making sure everything else is also up-to-date, once that's true I'll release a new version.

**Comment by GlowingUmbreon** [comment]
> I'm currently working on making sure everything else is also up-to-date, once that's true I'll release a new version.

Ah, do you not have a automatic system for doing so? I assumed that you had some script that would update enums.rs and instance.rs automatically.

**Comment by CraftSpider** [comment]
No, I intend to add one at some point, but don't currently have something that does it. At that point I would likely do something more like the `windows` crate does, and generate the actual definitions off of the API dumps. I do have a script that generates basically an instance of every type, and then I just run that through the tests and fix all the issues it surfaces.

**Comment by GlowingUmbreon** [comment]
Ah okay, also quick question how do you get the attributes for a instance? I have read the docs and looked at the source code but I still cannot figure it out. I assume I need to TryInto a actual instance something but I am sure how to do so.
```rs
    let settings = match map.get_path("0/Settings/Main") {
        Ok(settings) => settings,
        Err(ModelError::NotFound) => panic!("Couldn't find instance named \"Settings\""),
        Err(ModelError::AmbiguousPath) => panic!("Found more than one instance named \"Settings\""),
        Err(err) => panic!("Model Error: {}", err),
    };

    println!("{:?}", settings.attributes);
```

**Comment by CraftSpider** [comment]
`get_path` returns an `Instance`. To get the attributes, you would match on it - depending on what type you expect it to be, something like
```rs
match settings {
    Instance::Script(script) => { /* here, you access a script's attributes by name, such as `script.source` */ }
    _ => { /* What to do if the instance isn't what you expected */ }
}
```
I'm considering possible ways to simplify this a bit, for example providing a method on `Instance` to attempt to read it as an item or any subclass of that item. If you have any more questions, or ideas on how you'd prefer the API to look, feel free to open other issues. Also if there's any documentation you think should be added.
